source/sync/error-handling/client-resets.txt

```diff
@@ -82,121 +84,219 @@ with the client's history. The most common causes of client resets are:
-#. Performs a diff to figure out the set of steps required to
-   bring the original (local) copy of the realm to the same state as
-   the fresh copy of the backend.
-#. Applies that set of steps to transform the local realm into a state
-   where it can sync with the backend. This includes:
-
-   - deleting all local changes that haven't yet synced to the backend
-   - any inserts, updates, and deletes that haven't yet synced to the client
-
-#. Discards the fresh copy. The app continues to use
-   the original copy of the realm with the diff applied.
-
-The "discard unsynced changes" strategy has several advantages compared
-to manual recovery:
+This client reset mode has several advantages compared to manual recovery:
 
 - Your application can perform a client reset without writing any custom
-  logic, other than specifying this strategy. You don't have to manually
+  logic, other than specifying this mode. You don't have to manually
   initiate the client reset or interact with the error object at all.
 - Your application can perform this client reset without closing any
   realms, disconnecting from the backend app, or manual restarts. This
   means you don't have to write any logic to handle these situations.
 - Application users receive notifications for changes as the local realm
   updates to match the state of the backend realm.
 
-There is one major disadvantage to this strategy, however:
+There is one major disadvantage to this client reset mode, however:
 
 - There is no way to recover changes that have not yet synced to the
-  backend at the time of the client reset. When this strategy uses a
+  backend at the time of the client reset. When this mode uses a
   diff to bring the local realm to the same state as the backend, unsynced
   changes are permanently deleted and cannot ever be recovered.
 
-You can use this strategy to resolve a client reset caused by anything
-*except* a breaking schema change.
+Discard Unsynced Changes mode cannot perform an automated client reset in 
+the event of a breaking schema change.
 
+.. _manual-client-reset:
 .. _manually-recover-unsynced-changes:
 
-Manually Recover Unsynced Changes
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Manual Client Reset
+~~~~~~~~~~~~~~~~~~~
 
-.. note:: Manual Recovery Replaces Client Reset Handlers
+In the event of :ref:`a breaking schema change <breaking-change-quick-reference>`,
+Realm SDKs cannot automatically handle a client reset. You must define 
+a manual client reset handler if you make breaking schema changes.
 
-   This strategy works just like the deprecated
-   "Client Reset Handler" method of performing client resets.
-   Applications that use deprecated client reset handling can switch
-   directly to the "manually recover unsynced changes" strategy with
-   no logic changes.
+In this scenario, a manual client reset handler should do something like 
+tell the user to update the app. In Realm SDK versions that automatically 
```

Comment:
I don't understand that first sentence, why would a manual client reset handler tell the user to update the app? Seems like the developer should create code in this handler to update the objects in response to the breaking change. Am I misunderstanding something?

Reply:
We are assuming that you have made a breaking change intentionally, and you have released a new version of the app to handle the new models. Per engineering, the advice for the developer in this instance is to tell users to update to the new version of the app, since there is no meaningful way to recover from this type of error.